ios/RNSScreenStackComponentView.mm

```diff
@@ -269,8 +275,37 @@ - (void)dismissOnReload
 
 #pragma mark methods connected to transitioning
 
+- (void)cancelTouchesInParent
+{
+  // cancel touches in parent, this is needed to cancel RN touch events. For example when Touchable
+  // item is close to an edge and we start pulling from edge we want the Touchable to be cancelled.
+  // Without the below code the Touchable will remain active (highlighted) for the duration of back
+  // gesture and onPress may fire when we release the finger.
+  UIView *parent = _controller.view;
+  while (parent != nil && ![parent respondsToSelector:@selector(touchHandler)])
+    parent = parent.superview;
+
+  if (parent != nil) {
+    RCTTouchHandler *touchHandler = [parent performSelector:@selector(touchHandler)];
+    [touchHandler cancel];
+    [touchHandler reset];
+  }
+}
+
 - (BOOL)gestureRecognizerShouldBegin:(UIGestureRecognizer *)gestureRecognizer
 {
+  RNSScreenComponentView *topScreen = (RNSScreenComponentView *)_controller.viewControllers.lastObject.view;
+
+  if (topScreen.fullScreenSwipeEnabled) {
+    // we want only `RNSPanGestureRecognizer` to be able to recognize when
+    // `fullScreenSwipeEnabled` is set
+    if ([gestureRecognizer isKindOfClass:[RNSPanGestureRecognizer class]]) {
+      _isFullWidthSwiping = YES;
+      [self cancelTouchesInParent];
+      return YES;
+    }
+    return NO;
+  }
   return YES;
```

Comment:
Could you add here a comment like this `TODO: Add code for customAnimationOnSwipe`?

I think it will be useful and I think I should have added them in more places